### PR TITLE
Aakash_Review_Code

### PR DESCRIPTION
**Pull request by TheAakash1886**
Please review the code.
Thank you

### CODE REVIEW

File: Aakash_Irengbam_Collections.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import csv
+import string
+import Aakash_Irengbam_Dictionary as dic
+letter_occurence = dict( (key, [0,0,0,0,0]) for key in string.ascii_lowercase )
+def alphabet_likely():
+    f = open("wordListNew.txt", "r")
+    content = f.read()
+    WordsList = content.split('\n')
+    f.close()
```
Comment (jeffft9) [inline Aakash_Irengbam_Collections.py:9]: Exception Handling is missing for handling files, please enclose the code in "try except" blocks.

File: Aakash_Irengbam_Collections.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    myDict = {}
+    for line in range(1,len(tempList)):
+        temp = tempList[line].split(", ")
+        if len(temp[0]) != 0:
+            myDict[temp[0]] = (temp[1],temp[2],temp[3],temp[4],temp[5])
+    return myDict
+        
+def occur_likely():
+    f = open("wordListNew.txt", "r")
+    content = f.read()
```
Comment (jeffft9) [inline Aakash_Irengbam_Collections.py:46]: Mention working and purpose of function in one line in comments above the function for better understanding.

File: Aakash_Irengbam_Dictionary.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import random
+import os
+
+f = open("/Users/aakashirengbam/Downloads/words.txt", "r")       #open the text file
+wordlist = f.read()                                              #read the text file
+wordslist = wordlist.split("\n")
+f.close()
```
Comment (jeffft9) [inline Aakash_Irengbam_Dictionary.py:7]: Please add try-except blocks for exception handling

File: Aakash_Irengbam_Wordle.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+def main():                                                       #define the main function
+    RightWord = []                                                       
+    RightWord = Aakash_Irengbam_Dictionary.randomword()      #call to the dictionary to generate the random word
+    Aakash_Irengbam_Dictionary.removeWord(RightWord)
+    #HW03_Aakash_Irengbam_UI.userinterface(RightWord)              #send the generated random word to the User Interface
+    Inter = Aakash_Irengbam_UI.Interface(RightWord)
+    Inter.userinterface()
+    
+if __name__ == "__main__":                                        #run the main function
+    main()
```
Comment (jeffft9) [inline Aakash_Irengbam_Wordle.py:16]: No changes required. Well written code with comments.

File: Test.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import HW03_Aakash_Irengbam_Dictionary as Dictionary        #import the other modules to use their functions
+import HW03_Aakash_Irengbam_Wordle as Wordle
+import HW03_Aakash_Irengbam_UI as UI
+import HW07_Aakash_Irengbam_Collections as c
+import unittest                               #import the unittest
+from unittest.mock import patch
+
+class WordleTest(unittest.TestCase):
```
Comment (jeffft9) [inline Test.py:8]: Good number of test cases covered, well done.

File: WordleHelper.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+with open('wordRank.csv', newline='') as csvfile:
+    reader = csv.reader(csvfile, delimiter=' ', quotechar='|')
+    for row in reader:
+        rankDict[row[0].split(',')[0]] = row[1].split(',')[0]
+rankDict.pop('Rank')
+
+
+def rankedWords(correctLetters, incorrectLetters):
+    myList = []
+    defaultList = []
```
Comment (jeffft9) [inline WordleHelper.py:12]: Add a one line comment to describe working of the function

File: gameplay.log [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+RightWord: truth
+Guess: sales
+RightWord: quest
+Guess: sales
+RightWord: bases
+Guess: sales
+RightWord: limit
+Guess: sales
+RightWord: aware
+Guess: sales
```
Comment (jeffft9) [inline gameplay.log:114]: Guess distribution for the game is missing, to show the attempts taken to complete the games.

Comment (jeffft9) [reply gameplay.log:114]: After playing multiple games you can display distribution as [3,5,"Unsuccessful",4] indicating the attempts for total 4 games.

File: Aakash_Irengbam_UI.py [hunk 8]
```diff
@@ -0,0 +1,254 @@
+import Aakash_Irengbam_Dictionary
```
Comment (jeffft9) [inline Aakash_Irengbam_UI.py:1]: Try using an alias so you dont have to write the whole filename for access its methods. 
Example : import Aakash_Irengbam_Dictionary as wordDict

**Review by jeffft9 — CHANGES_REQUESTED** [review]
Please add unnecessary files to ".gitignore" like the directory "__pycache__" because it is not required to be uploaded to github.

File: Aakash_Irengbam_Collections.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    f.write("letter,first_pos, second_pos, third_pos, fourth_pos, fifth_pos \n")
+    for letter in string.ascii_lowercase:
+        f.write(f"{letter}, {letter_occurence[letter][0]}, {letter_occurence[letter][1]}, {letter_occurence[letter][2]}, {letter_occurence[letter][3]}, {letter_occurence[letter][4]}\n")
+    f.close()
+
+def Convert_To_tuple(dic):
+    for key in dic.keys():
+        dic[key] = tuple(dic[key])
+
+def ParseToTuples():
```
Comment (aishwaryashirbhate) [inline Aakash_Irengbam_Collections.py:33]: Please add in type hints for the functions.

File: Aakash_Irengbam_Dictionary.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    if len(x) == 5:                                              #only read text file if the length of the letter is 5 words
+        #newList.append(x.lower())
+        f.write(f"{x.lower()}\n")
+f.close()
+f = open("wordListNew.txt", "r")
+content = f.read()
+newList = content.split("\n")
+        
+        
+def randomword():                                                #choose a random word from the dictionary
```
Comment (aishwaryashirbhate) [inline Aakash_Irengbam_Dictionary.py:19]: Please add function definition and type hints for all the function stating its purpose and working.

File: Test.py [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    
+    def test_file_status_funciton(self) -> None :
+        """Testing the file status function"""
+        self.assertFalse(Dictionary.fileStatus())
+    
+    def test_alphabet_function_false(self) -> None :
+        """Testing alphabet likely function for false value"""
+        self.assertFalse(c.checkLen(""))
+        
+    def test_alphabet_function_true(self) -> None :
```
Comment (aishwaryashirbhate) [inline Test.py:57]: Good job in using relevant function names as per test case.

File: WordleSolver.py [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+# wordslist = wordlist.split("\n")
+# f.close()
+# f = open("HelperWordList.txt", "w")
+# for x in wordslist:
+#     if len(x) == 5:                                              #only read text file if the length of the letter is 5 words
+#         f.write(f"{x.lower()}\n")
+# f.close()
+Rnd = dict.randomword()
+UI = UI.Interface(Rnd)
+def Solver(Betterattempts):
```
Comment (aishwaryashirbhate) [inline WordleSolver.py:17]: Please add type hints and function definition.

**Review by aishwaryashirbhate — CHANGES_REQUESTED** [review]

**Pull request merged**